zerver/lib/create_user.py

```diff
@@ -81,12 +81,17 @@ def create_user(email: str, password: Optional[str], realm: Realm,
                 bot_type: Optional[int] = None,
                 bot_owner: Optional[UserProfile] = None,
                 tos_version: Optional[str] = None, timezone: str = "",
+                user_avatar_url: str='',
                 avatar_source: str = UserProfile.AVATAR_FROM_GRAVATAR,
                 is_mirror_dummy: bool = False,
                 default_sending_stream: Optional[Stream] = None,
                 default_events_register_stream: Optional[Stream] = None,
                 default_all_public_streams: Optional[bool] = None,
                 source_profile: Optional[UserProfile] = None) -> UserProfile:
+    # If a user is being created after authenticating using GitHub, than it doesn't make
+    # sense to have avatar_source == UserProfile.AVATAR_FROM_GRAVATAR and also
+    # user_avatar_url as their GitHub avatar url.
```

Comment:
same here, right now it is indeed a github avatar url, but it can be anything, and likely will be